### PR TITLE
Hide sriov ecap for ptdev

### PR DESCRIPTION
**Pull request by taoyuhong**

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
2 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Implicit conversion: actual to formal param (MR). :  (struct acrn_vpci):  param  1<br><2> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Logical conjunctions need brackets.<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR,thanks

**Comment by HuiHuangShi** [comment]
1 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Implicit conversion: actual to formal param (MR). :  (struct acrn_vpci):  param  1<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
1 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Casting operation to a pointer. : (struct acrn_vpci to Unknown_Type:acrn_vpci*): ( const struct acrn_vpci * ) ( vdev -> vpci )<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
2 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Implicit conversion: actual to formal param (MR). :  (struct acrn_vpci):  param  1<br><2> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Logical conjunctions need brackets.<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by shiqingg** [comment]
Add 'Coding Guidelines: PASS' label on behalf of @HuiHuangShi

File: hypervisor/dm/vpci/vpci.c [hunk 1]
```diff
@@ -495,8 +495,10 @@ static int32_t write_pt_dev_cfg(struct pci_vdev *vdev, uint32_t offset,
 		write_sriov_cap_reg(vdev, offset, bytes, val);
 	} else {
 		if (!is_quirk_ptdev(vdev)) {
-			/* passthru to physical device */
-			pci_pdev_write_cfg(vdev->pdev->bdf, offset, bytes, val);
+			if (offset != vdev->pdev->sriov.pre_pos) {
+				/* passthru to physical device */
+				pci_pdev_write_cfg(vdev->pdev->bdf, offset, bytes, val);
+			}
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c:501]: Lifei said this is incorrect. We need to do this before if (!is_quirk_ptdev()..
I am fine with either. Please check with Fei, and ask him to review.

File: hypervisor/dm/vpci/vpci.c [hunk 2]
```diff
@@ -520,8 +522,12 @@ static int32_t read_pt_dev_cfg(const struct pci_vdev *vdev, uint32_t offset,
...
 	} else {
 		if (!is_quirk_ptdev(vdev)) {
-			/* passthru to physical device */
-			*val = pci_pdev_read_cfg(vdev->pdev->bdf, offset, bytes);
+			if (offset == vdev->pdev->sriov.pre_pos) {
+				*val = pci_vdev_read_vcfg(vdev, offset, bytes);
+			} else {
+				/* passthru to physical device */
+				*val = pci_pdev_read_cfg(vdev->pdev->bdf, offset, bytes);
+			}
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c:530]: same thing here

File: hypervisor/dm/vpci/vpci.c (hunk 1 above)
Comment (taoyuhong) [reply hypervisor/dm/vpci/vpci.c:501]: Li Fei has see the PR patch, he reviewed and ask me to PR, I will ask him for this issue

Comment (taoyuhong) [reply hypervisor/dm/vpci/vpci.c:501]: He said should before  if (!is_quirk_ptdev(), will fix that

File: hypervisor/dm/vpci/vpci.c (hunk 2 above)
Comment (taoyuhong) [reply hypervisor/dm/vpci/vpci.c]: will fix that

**Comment by HuiHuangShi** [comment]
2 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Implicit conversion: actual to formal param (MR). :  (struct acrn_vpci):  param  1<br><2> config:common filename:/hypervisor/dm/vpci/pci_pt.c func:init_vdev_pt offset:422<br> reason:Logical conjunctions need brackets.<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/dm/vpci/vpci.c (hunk 2 above)
Comment (taoyuhong) [reply hypervisor/dm/vpci/vpci.c]: fixed

File: hypervisor/dm/vpci/vpci.c (hunk 1 above)
Comment (taoyuhong) [reply hypervisor/dm/vpci/vpci.c:503]: fixed

**Pull request merged**